lib/spack/docs/module_file_support.rst

```diff
@@ -553,6 +573,9 @@ most likely via the ``+blas`` variant specification.
      packages that satisfy a spec in ``core_specs`` will be put in the
      ``Core`` hierarchy of the lua modules.
 
+     It is highly recommended NOT to include dependencies in the lmod
+     hierarchy in the projections for lmod modules.
```

Comment:
Can you make this sentence a bit clearer?  Maybe "When specifying module names this way, we recommend NOT including names of dependencies (e.g., MPI, compilers) that are already in the `lmod` hierarchy."